### PR TITLE
Migration of pipeline to AWS Concourse

### PR DESCRIPTION
**Pull request by liamraemclauchlan**
* Use new PR resource
* Change target to aws-concourse
* Change secrets
* Remove aws credentials for terraform (including assume_role)

### CODE REVIEW

File: aviator.yml [hunk 1]
```diff
@@ -12,6 +12,6 @@ spruce:
   to: aviator_pipeline.yml
 fly:
   name: orchestration-service
-  target: concourse
-  expose: false
+  target: aws-concourse
+  expose: true
```
Comment (dan-hill2802) [inline aviator.yml:16]: can you also add `check_creds: true`? I'll put a full explanation to the team

File: ci/meta.yml [hunk 2]
```diff
@@ -24,26 +24,27 @@ meta:
...
         platform: linux
         image_resource:
           type: docker-image
           source:
-            repository: ((terraform.repository))
-            version: ((terraform12.version))
-            tag: ((terraform12.version))
+            repository: hashicorp/terraform
+            version: ((dataworks.terraform_version))
+            tag: ((dataworks.terraform_version))
```
Comment (dan-hill2802) [inline ci/meta.yml:37]: you don't need `tag` in newer Concourse, just `version

File: ci/meta.yml [hunk 3]
```diff
@@ -24,26 +24,27 @@ meta:
...
+    terraform-common-config:
       config:
         platform: linux
         image_resource:
           type: docker-image
           source:
-            repository: ((terraform.repository))
-            version: ((terraform12.version))
-            tag: ((terraform12.version))
+            repository: hashicorp/terraform
```
Comment (dan-hill2802) [inline ci/meta.yml:35]: is the repo not in secretes manager?

File: ci/resources.yml [hunk 4]
```diff
@@ -2,16 +2,17 @@ resources:
 - name: orchestration-service-pr
   type: pull-request
   source:
-    access_token: ((github_com.token))
+    access_token: ((dataworks-secrets.enterprise_github_pat))
     repo: dwp/orchestration-service
     uri: https://github.com/dwp/orchestration-service.git
-  check_every: 15m
+  check_every: 5m
```
Comment (dan-hill2802) [inline ci/resources.yml:8]: you should set the check to `720h` and add the webhook in `dataworks-github-config`

File: ci/resources.yml [hunk 5]
```diff
@@ -2,16 +2,17 @@ resources:
...
+
 - name: orchestration-service
   type: git
   source:
     branch: master
-    username: ((github_com.username))
-    password: ((github_com.token))
+    access_token: ((dataworks-secrets.enterprise_github_pat))
     uri: https://github.com/dwp/orchestration-service.git
   check_every: 15m
```
Comment (dan-hill2802) [inline ci/resources.yml:17]: you should set the check to `720h` and add the webhook in `dataworks-github-config`

File: ci/meta.yml (hunk 2 above)
Comment (dan-hill2802) [reply ci/meta.yml:37]: for reference: https://concourse-ci.org/tasks.html#schema.anonymous_resource

Comment (liamraemclauchlan) [reply ci/meta.yml:36]: I think we need tag, but not version (from what I can see in the docs)

File: aviator.yml (hunk 1 above)
Comment (liamraemclauchlan) [reply aviator.yml:16]: Added

File: ci/meta.yml (hunk 3 above)
Comment (liamraemclauchlan) [reply ci/meta.yml:35]: No idea, just following aws-metrics-infrastructure and aws-management-infrastructure

Comment (ChrisScottThomas) [reply ci/meta.yml:35]: yes it is. `((dataworks.terraform_repository))` But it doesn't feel like it should be a hard requirement

Comment (benjaminsherwood) [reply ci/meta.yml:35]: Yea, I was also copying previous ones, but maybe it _should_ be using the secrets one if it's there, same with mine.

Comment (dan-hill2802) [reply ci/meta.yml:35]: it enables us to move all references quickly when Docker Hub is not working

Comment (benjaminsherwood) [reply ci/meta.yml:35]: Yea, agreed. I've asked @liamraemclauchlan to update it. He copied it from a source we hadn't updated.

Comment (ChrisScottThomas) [reply ci/meta.yml:35]: Potentially, but as it's a non-private/secret parameter, I don't have any feelings either way.  
As a side bar, if you want to know how to quest SM to get the list of stored params, give me a prod on Slack.

**Pull request merged**